state-chain/custom-rpc/src/lib.rs

```diff
@@ -1093,7 +1129,20 @@ where
 	) -> Result<(), SubscriptionEmptyError> {
 		let from = from_asset.try_into().map_err(|_| SubscriptionEmptyError)?;
 		let to = to_asset.try_into().map_err(|_| SubscriptionEmptyError)?;
-		self.new_items_subscription(sink, move |api, hash| api.cf_prewitness_swaps(hash, from, to))
+		self.new_items_subscription(sink, move |api, hash, block_number| {
```

Comment:
Can you not factor this mapping out into the new_items_subscription and new_update_subscription functions?

Reply:
This is very nice - we no longer need to pass blocknumber into here, and we no longer need the "withContent" trait since we have access to the inside content within the `new_items_subscription` function